whelktool/src/main/groovy/datatool/WhelkTool.groovy

```diff
@@ -355,6 +360,35 @@ class WhelkTool {
         }
     }
 
+    private boolean doRevertToTime(DocumentItem item) {
+
+        // The 'versions' list is sorted, with the most recent version first.
+        List<Document> versions = whelk.storage.loadAllVersions(item.doc.shortId)
+
+        ZonedDateTime restoreTime = ZonedDateTime.parse(item.restoreToTime)
+        Document selectedVersion = versions.get(versions.size()-1)
+
+        // Go over the versions, oldest first (in reverse),
+        // until you've found the oldest version that is younger than the desired time target.
```

Comment:
Should we silently ignore cases where `restoreTime` is older than any version timestamp or do we need to inform the user?

Reply:
I've pushed a new version, logging such occurrences.